src/config/Config.cpp

```diff
@@ -265,4 +265,27 @@ void Config::SetWindowBackend(WindowBackend backend) {
     }
 }
 
+bool Config::RegisterConfigVersionUpdater(std::shared_ptr<ConfigVersionUpdater> versionUpdater) {
+    auto [_, emplaced] = mVersionUpdaters.emplace(versionUpdater->GetVersion(), versionUpdater);
+    return emplaced;
+}
+
+void Config::RunVersionUpdates() {
+    for (auto [_, versionUpdater] : mVersionUpdaters) {
+        uint32_t version = GetUInt("ConfigVersion", 0);
+        if (version == versionUpdater->GetVersion()) {
+            versionUpdater->Update(this);
+            SetUInt("ConfigVersion", version + 1);
+        }
+    }
```

Comment:
```suggestion
    for (auto [_, versionUpdater] : mVersionUpdaters) {
        uint32_t version = GetUInt("ConfigVersion", 0);
        if (version < versionUpdater->GetVersion()) {
            versionUpdater->Update(this);
            SetUInt("ConfigVersion", versionUpdater->GetVersion());
        }
    }
```

Reply:
This looks like the correct logic to me. This can allow a port to do something like using version codes in the config version.

Reply:
Updated